### ISSUE TITLE
[GSoC 2021 Proposal] Few shot algorithms in JAX

### ISSUE THREAD

**Issue by rakesh4real** [google/jax#5802]
Hi!

I want to implement an end-to-end framework for few/one shot learning algorithms using jax.
This is will be a long term project but currently, I am planning to implement the following as a part of GSoC 2021

- Siamese network with contrastive loss and triplet loss
- Prototype Network
- Matching Network

Any references to how to proceed will be immensely helpful.

Cheers!
Rakesh

**Comment by mattjj** [comment]
Hey @rakesh4real, that sounds really fun!

I suggest checking out the Flax or Haiku libraries, as you'll likely want to use more than just core JAX to build and train these kinds of models.

Hope your project goes well!
